src/main/java/com/minecolonies/coremod/network/messages/TransferItemsRequestMessage.java

```diff
@@ -0,0 +1,154 @@
+{
+    /**
+     * The id of the building.
+     */
+    private BlockPos buildingId;
+    /**
+     * The id of the colony.
+     */
+    private int      colonyId;
+    /**
+     * How many item need to be transfer from the player inventory to the building chest
+     */
+    private ItemStack      itemStack;
+    /**
+     * How many item need to be transfer from the player inventory to the building chest
+     */
+    private int      quantity;
+
+    /**
+     * Empty constructor used when registering the message.
+     */
+    public TransferItemsRequestMessage()
+    {
+
+        super();
+    }
+
+    /**
+     * Creates a Transfer Items request message.
+     *
+     * @param building AbstractBuilding of the request.
+     * @param itemStack to be take from the player for the building
+     * @param quantity of item needed to be transfered
+     */
+    public TransferItemsRequestMessage(@NotNull final AbstractBuilding.View building, final ItemStack itemStack, final int quantity)
+    {
+        super();
+        this.colonyId   = building.getColony().getID();
+        this.buildingId = building.getID();
+        this.itemStack  = itemStack;
+        this.quantity   = quantity;
+
+    }
+
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        colonyId   = buf.readInt();
+        buildingId = BlockPosUtil.readFromByteBuf(buf);
+        itemStack  = ByteBufUtils.readItemStack(buf);
+        quantity   = buf.readInt();
+    }
+
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        buf.writeInt(colonyId);
+        BlockPosUtil.writeToByteBuf(buf, buildingId);
+	ByteBufUtils.writeItemStack(buf,itemStack);
+        buf.writeInt(quantity);
+    }
+
+    @Override
+    public void messageOnServerThread(final TransferItemsRequestMessage message, final EntityPlayerMP player)
+    {
+
+        final Colony colony = ColonyManager.getColony(message.colonyId);
+        if (colony == null)
+        {
+            Log.getLogger().warn("TransferItemsRequestMessage colony is null");
+            return;
+        }
+
+        final AbstractBuilding building = colony.getBuilding(message.buildingId);
+        if (building == null)
+        {
+            Log.getLogger().warn("TransferItemsRequestMessage building is null");
+            return;
+        }
+
+        if (message.quantity <= 0)
+        {
+            Log.getLogger().warn("TransferItemsRequestMessage quantity below 0");
+            return;
+        }
+
+        final Item item = message.itemStack.getItem();
+        final int amountInPlayer = InventoryUtils.getItemCountInInventory(player.inventory, item, message.itemStack.getMetadata());
+        final int amountToTake = Math.min(message.quantity, InventoryUtils.getItemCountInInventory(player.inventory, item, message.itemStack.getItemDamage()));
+        ItemStack itemStackToTake = new ItemStack(item, amountToTake, message.itemStack.getItemDamage());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'itemStackToTake' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)